Exclude -apple-pay-button from applying to any element that supports appearance: auto and is not a button

pxlcoder opened the pull request:
#### 75f3ded86715b6a6ab4c4e750a005fae5f9b8e8e
<pre>
Exclude -apple-pay-button from applying to any element that supports appearance: auto and is not a button
<a href="https://bugs.webkit.org/show_bug.cgi?id=272366">https://bugs.webkit.org/show_bug.cgi?id=272366</a>
<a href="https://rdar.apple.com/126107516">rdar://126107516</a>

Reviewed by Tim Nguyen.

`appearance: -apple-pay-button` is non-standard, and must be maintained for
compatibility. However, it is not necessary for elements that already have an
`auto` appearance, and are not buttons, to support the non-standard value.

* LayoutTests/TestExpectations:
* LayoutTests/fast/css/appearance-apple-pay-button-div-expected.html: Added.
* LayoutTests/fast/css/appearance-apple-pay-button-div.html: Added.
* LayoutTests/fast/css/appearance-apple-pay-button-inapplicable-expected.html: Added.
* LayoutTests/fast/css/appearance-apple-pay-button-inapplicable.html: Added.
* LayoutTests/fast/css/appearance-apple-pay-button-input-type-button-expected.html: Added.
* LayoutTests/fast/css/appearance-apple-pay-button-input-type-button.html: Added.
* LayoutTests/platform/ios/TestExpectations:
* LayoutTests/platform/mac/TestExpectations:
* Source/WebCore/rendering/RenderTheme.cpp:
(WebCore::RenderTheme::adjustAppearanceForElement const):

Ensure that `appearance: -apple-pay-button` remains supported on buttons and
element&apos;s that do not have an `auto` appearance, like `&lt;div&gt;`.

`&lt;input&gt;` with no `auto` appearance are excluded. This covers the `hidden`,
`file`, and `image` types, which have no requirement to support

`appearance: -apple-pay-button`.
Canonical link: <a href="https://commits.webkit.org/277268@main">https://commits.webkit.org/277268@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/8b4ddf6e863917443ed7d28e4b3e2a81941c0b06

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/47161 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/26340 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/49804 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/49844 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/43210 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/49468 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/31653 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/23797 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/38414 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/47742 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/47/builds/23831 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/40645 "Passed tests") | [❌ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/19724 "Found 1 new API test failure: /WPE/TestResources:/webkit/WebKitWebView/sync-request-on-max-conns (failure)") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/21239 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/41798 "Passed tests") | [✅ 🛠 wpe-skia](https://ews-build.webkit.org/#/builders/52/builds/5205 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/43548 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/42204 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/51720 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/22186 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/50/builds/18565 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/45718 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/23463 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/40791 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/44725 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/10400 "Built successfully and passed tests") | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/43/builds/24248 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/23180 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

nt1m commented on LayoutTests/fast/css/appearance-apple-pay-button-div-expected.html:
```diff
@@ -0,0 +1,51 @@
+    }
+
+    .apple-pay-button-div {
+        display :block;
```
Weird spacing

nt1m commented on LayoutTests/fast/css/appearance-apple-pay-button-div.html:
```diff
@@ -0,0 +1,52 @@
+    }
+
+    .apple-pay-button-div {
+        display :block;
```
Ditto

nt1m commented on LayoutTests/fast/css/appearance-apple-pay-button-input-type-button-expected.html:
```diff
@@ -0,0 +1,51 @@
+    }
+
+    .apple-pay-button-div {
+        display :block;
```
Ditto

nt1m commented on LayoutTests/fast/css/appearance-apple-pay-button-input-type-button.html:
```diff
@@ -0,0 +1,53 @@
+    }
+
+    .apple-pay-button-input {
+        display :block;
```
ditto

nt1m commented on Source/WebCore/rendering/RenderTheme.cpp:
```diff
@@ -158,6 +158,19 @@ StyleAppearance RenderTheme::adjustAppearanceForElement(RenderStyle& style, cons
     }
 
+#if ENABLE(APPLE_PAY)
+    if (appearance == StyleAppearance::ApplePayButton) {
```
Maybe add a comment to say it is only to apply -apple-pay-button on buttons and non-form controls.

pxlcoder committed:
Exclude -apple-pay-button from applying to any element that supports appearance: auto and is not a button
https://bugs.webkit.org/show_bug.cgi?id=272366
rdar://126107516

Reviewed by Tim Nguyen.

`appearance: -apple-pay-button` is non-standard, and must be maintained for
compatibility. However, it is not necessary for elements that already have an
`auto` appearance, and are not buttons, to support the non-standard value.

* LayoutTests/TestExpectations:
* LayoutTests/fast/css/appearance-apple-pay-button-div-expected.html: Added.
* LayoutTests/fast/css/appearance-apple-pay-button-div.html: Added.
* LayoutTests/fast/css/appearance-apple-pay-button-inapplicable-expected.html: Added.
* LayoutTests/fast/css/appearance-apple-pay-button-inapplicable.html: Added.
* LayoutTests/fast/css/appearance-apple-pay-button-input-type-button-expected.html: Added.
* LayoutTests/fast/css/appearance-apple-pay-button-input-type-button.html: Added.
* LayoutTests/platform/ios/TestExpectations:
* LayoutTests/platform/mac/TestExpectations:
* Source/WebCore/rendering/RenderTheme.cpp:
(WebCore::RenderTheme::adjustAppearanceForElement const):

Ensure that `appearance: -apple-pay-button` remains supported on buttons and
element's that do not have an `auto` appearance, like `<div>`.

`<input>` with no `auto` appearance are excluded. This covers the `hidden`,
`file`, and `image` types, which have no requirement to support

`appearance: -apple-pay-button`.
Canonical link: https://commits.webkit.org/277268@main
